src/Illuminate/Database/Query/Builder.php

```diff
@@ -682,7 +682,7 @@ public function where($column, $operator = null, $value = null, $boolean = 'and'
         // If the column is making a JSON reference we'll check to see if the value
         // is a boolean. If it is, we'll add the raw boolean string as an actual
         // value to the query to ensure this is properly handled by the query.
-        if (Str::contains($column, '->') && is_bool($value)) {
+        if (Str::contains((string) $column, '->') && is_bool($value)) {
```

Comment:
Personally I would probably use the following so we only skipped specific condition:

```suggestion
        if (Str::contains($column instanceof Expression ? (string) $column : $column, '->') && is_bool($value)) {
```

Reply:
Thanks @crynobone